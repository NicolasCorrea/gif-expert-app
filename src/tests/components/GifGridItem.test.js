import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('<GifGridItem /> Tests', () => {
    const title = "gif title";
    const url = "https://www.git.url";
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    test('should render component', () => {
        
        
        
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should have a paragraph with title', () => {
        const p = wrapper.find("p");
        expect(p.text().trim()).toBe(title)
    })
    
    test('should have image equal to url and alt from props', () => {
        const image = wrapper.find("img");
        
        expect(image.prop("src")).toBe(url);
        expect(image.prop("alt")).toBe(title);
    })
    
    test('should have class animated_fadeIn', () => {
        const div = wrapper.find("div");
        
        expect(div.prop("className").includes("animate__fadeIn")).toBe(true)
    })
    
    
    
})