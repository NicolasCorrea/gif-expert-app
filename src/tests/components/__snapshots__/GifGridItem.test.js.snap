// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> Tests should render component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="gif title"
    src="https://www.git.url"
  />
  <p>
    gif title
  </p>
</div>
`;
